### PR TITLE
test: authorised key testing

### PR DESCRIPTION
**Pull request by tlm**
This PR adds BASH tests for asserting authorised key functionality. We are doing this so as we move the CRUD and authorised keys functions over to DQlite in 4.0 we can have some certainty around how things operate and that the core contracts have been maintained.

This PR is required by #17602 

I also made a quick change to the authorized key worker so that logging on key changes is always at info now and not debug. @hpidcock and myself believed that this should always be info for security auditing purposes.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- ~[ ] Go unit tests, with comments saying what you're testing~
- [x] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```bash
cd tests
./main.sh authorizedkeys
```

## Documentation changes

N/A

## Links

**Jira card:** JUJU-6342

### CODE REVIEW

File: tests/suites/authorized_keys/task.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+test_authorized_keys() {
```
Comment (wallyworld) [inline tests/suites/authorized_keys/task.sh:1]: make consistent. "authorised" etc

**Review by wallyworld — APPROVED** [review]
Please have the test turn on debug logging for the worker and not make a unilateral change to the logging just for the test

File: worker/authenticationworker/worker.go [hunk 2]
```diff
@@ -123,8 +123,8 @@ func (kw *keyupdaterWorker) Handle(_ <-chan struct{}) error {
 	deleted := kw.jujuKeys.Difference(newJujuKeys)
 	added := newJujuKeys.Difference(kw.jujuKeys)
 	if added.Size() > 0 || deleted.Size() > 0 {
-		logger.Debugf("adding ssh keys to authorised keys: %v", added)
-		logger.Debugf("deleting ssh keys from authorised keys: %v", deleted)
+		logger.Infof("adding ssh keys to authorised keys: %v", added)
```
[the remark this answers is not in the document]
Comment (tlm) [reply worker/authenticationworker/worker.go:126]: No sorry I should have explained this change in the PR. I will update it now. @hpidcock and myself made the call to make these info as they are security related log messages and re think they should always be surfaced for auditing purposes.

File: tests/suites/authorized_keys/task.sh (hunk 1 above)
Comment (tlm) [reply tests/suites/authorized_keys/task.sh:1]: Done & Done

**Pull request merged**